packages/supermassive/src/codegen/__tests__/index.ts

```diff
@@ -0,0 +1,684 @@
+        export declare namespace User {
+            export type id = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
+            export type name = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
+        }
+        export declare namespace Query {
+            export type users = (model: unknown, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<(UserModel | null)[] | null>;
+        }
+        "
+      `);
+    });
+
+    test("if a type is not used it still need to be imported in resolvers", () => {
+      let { resolvers, models } = runGenerateTest(graphql`
+        interface Node {
+          id: ID!
+        }
+
+        type User implements Node {
+          id: ID!
+          name: String!
+        }
+      `);
+      expect(models).toMatchInlineSnapshot(`
+        "export type BaseModel = {
+            __typename: string;
+        };
+        export interface NodeModel extends BaseModel {
+            __typename: string;
+        }
+        export interface UserModel extends BaseModel, NodeModel {
+            __typename: \\"User\\";
+            id: string;
+            name: string;
+        }
+        "
+      `);
+      expect(resolvers).toMatchInlineSnapshot(`
+        "import type { PromiseOrValue } from \\"@graphitation/supermassive\\";
+        import type { ResolveInfo } from \\"@graphitation/supermassive\\";
+        import type { NodeModel, UserModel } from \\"./models.interface.ts\\";
+        export declare namespace User {
+            export type id = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
+            export type name = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
+        }
+        "
+      `);
+    });
+
+    test("Input", () => {
+      let { resolvers, models } = runGenerateTest(graphql`
+        type User {
+          id: ID!
+        }
+
+        input UserParamsInput {
+          name: String
+        }
+
+        extend type Query {
+          userById(params: UserParamsInput): User
+        }
+      `);
+      expect(models).toMatchInlineSnapshot(`
+        "export type BaseModel = {
+            __typename: string;
+        };
+        export interface UserModel extends BaseModel {
+            __typename: \\"User\\";
+            id: string;
+        }
+        "
+      `);
+      expect(resolvers).toMatchInlineSnapshot(`
+        "import type { PromiseOrValue } from \\"@graphitation/supermassive\\";
+        import type { ResolveInfo } from \\"@graphitation/supermassive\\";
+        import type { UserModel, UserParamsInputModel } from \\"./models.interface.ts\\";
+        export declare namespace User {
+            export type id = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
+        }
+        export type UserParamsInput = {
```

Comment:
Why?

Reply:
@vladar and I decided that the input should be in the resolvers file, since it's related to resolvers directly and not to models.